src/ui/component/field/numericalField.tsx

```diff
@@ -0,0 +1,69 @@
+import { useCallback, useMemo } from 'react'
+import * as O from 'fp-ts/Option'
+import * as S from 'fp-ts/string'
+import { pipe, flow } from 'fp-ts/function'
+import type { FieldProps } from '@/ui/component/field/field'
+import { Field } from '@/ui/component/field/field'
+import { isValidNumberFieldFormat } from '@/util/field/field'
+import { isEmptyOrEndsWithDot } from '@/util/util'
+import { formatLocalizedNumber } from '@/util/i18n/i18n'
+
+type NumericalFieldProps = FieldProps & {
+  onChange: (value: string) => void
+}
```

Comment:
I don't like too much this because it creates a duplicate definition of the `onChange` prop, which could lead to maintenance issues if the `FieldProps` definition changes.

I'd like to suggest this change:

```suggestion
type NumericalFieldProps = FieldProps & Required<Pick<FieldProps, 'onChange'>>;
```

This allows all changes to be inherited from the `onChange` prop in `FieldProps`, ensuring that both types remain in sync, imho.

Reply:
Beautiful! ✨ Thanks!